src/features/DocumentFormatter.ts

```diff
@@ -260,14 +260,23 @@ class PSDocumentFormattingEditProvider implements
         this.lockDocument(document, textEdits);
         PSDocumentFormattingEditProvider.showStatusBar(document, textEdits);
 
-        return textEdits.then(
-            (edits) => {
-                this.logger.writeVerbose(`Document formatting finished in ${getFormattingDuration()}s`);
-                return edits;
-            },
-            (err) => {
-                this.logger.writeVerbose(`Document formatting failed in ${getFormattingDuration()}: ${err}`);
-            });
+        return this.logAndReturnTextEdits(textEdits, getFormattingDuration);
+    }
+
+    // There is something about having this code in the calling method that causes a TS compile error.
+    // It causes the following error:
+    // Type 'import("C:/Users/Keith/GitHub/rkeithhill/vscode-powershell/node_modules/vscode-languageserver-typ...'
```

Comment:
Is this the issue you were hitting that you fixed with `npm run postinstall`?

Reply:
No, that was another issue.  The one above happened when I simply changed from TS 2.3.x to 2.4.x - bam I'd get that error.  My best guess is that newer versions of TS detect the issue.  The nice thing about TS 2.9.1 was that it had a code fix for the issue - which I applied above.

Reply:
I tried playing around with this just to see if the type error meant anything... Between `vscode`, `vscode-languageclient` and `vscode-languageserver-types` there seems to be a whole can of worms that I can't find much documentation of. So I think this solution is the most tenable for now.